api/types/src/derives.rs

```diff
@@ -240,6 +240,36 @@ impl_poem_type!(
     )
 );
 
+impl_poem_type!(
+    U16,
+    "string",
+    (
+        example = Some(serde_json::Value::String("65534".to_string())),
+        format = Some("uint64"),
+        description = Some(indoc! {"
+      A string containing a 16-bit unsigned integer.
+
+      We represent u16 values as a string to ensure compatibility with languages such
+      as JavaScript that do not parse u64s in JSON natively.
+  "})
+    )
+);
+
+impl_poem_type!(
+    U32,
+    "string",
+    (
+        example = Some(serde_json::Value::String("2147483647".to_string())),
+        format = Some("uint64"),
+        description = Some(indoc! {"
+    A string containing a 32-bit unsigned integer.
+
+    We represent u32 values as a string to ensure compatibility with languages such
+    as JavaScript that do not parse u64s in JSON natively.
```

Comment:
How come we have these? Currently we only have U64 and up because JS can only handle U32 or lower natively. Meaning in this case, we should be able to just use the numbers directly instead of having this string wrapper type.

Reply:
Or perhaps we still need U32, but not U16.

Reply:
Thanks for catching us. I've deleted u16 and u32. JS uses float64 to represent numbers and should be able to handle integers between -2^53 and 2^53, so u32 should be fine.